rocky/katalogus/views/plugin_detail.py

```diff
@@ -144,54 +174,60 @@ def get_context_data(self, **kwargs):
         return context
 
     def post(self, request, *args, **kwargs):
-        """Start scanning oois at plugin detail page."""
-        if not self.indemnification_present:
-            return self.get(request, *args, **kwargs)
+        action = request.POST["action"]
 
-        if "boefje_id" not in request.POST:
-            raise BadRequest("No boefje_id provided")
-
-        selected_oois = request.POST.getlist("ooi")
-        plugin_id = request.POST["boefje_id"]
-        if selected_oois and plugin_id:
-            oois = self.get_oois(selected_oois)
-            boefje = self.katalogus_client.get_plugin(plugin_id)
-
-            oois_with_clearance_level = oois["oois_with_clearance"]
-            oois_without_clearance_level = oois["oois_without_clearance"]
-
-            if oois_with_clearance_level:
-                self.run_boefje_for_oois(
-                    boefje=boefje,
-                    oois=oois_with_clearance_level,
-                )
-
-            if oois_without_clearance_level:
-                if not self.organization_member.has_perm("tools.can_set_clearance_level"):
-                    messages.add_message(
-                        self.request,
-                        messages.ERROR,
-                        _(
-                            "Some selected OOIs needs an increase of clearance level to perform scans."
-                            " You do not have the permission to change clearance level."
-                        ),
+        if action == PageActions.RESCHEDULE_TASK.value:
+            self.handle_page_action(action)
+            return redirect(request.path)
+        else:
```

Comment:
This else is not needed, the whole indented piece can only be reached if the condition above is not true because of the return in there.